usl_pipeline/usl_lib/usl_lib/readers/polygon_readers.py

```diff
@@ -0,0 +1,49 @@
+import typing
+from typing import Tuple
+
+from shapely import geometry
+
+
+def read_polygons_from_text_file(
+    file: typing.TextIO, support_mask_values: bool = False
+) -> list[Tuple[geometry.Polygon, int]]:
+    """Reads polygon information with optional mask values from text file.
+
+    Args:
+        file: File text stream to load from.
+        support_mask_values: Optional indicator of the presence of polygon mask values
+            in the first column in polygon lines. Default case is that mask column is
+            not present (in this case mask value 1 will be assigned to every polygon).
+
+    Returns:
+        The list of tuples combining polygon with associated mask.
+    """
+    lines = file.readlines()
+
+    # First line contains the number of polygon lines
+    polygon_count = int(lines[0])
+    polygons = []
+
+    # The rest of lines have structure of white-space separated numeric values, where
+    # first optional column has mask value (if this mode was requested by caller), then
+    # the number of points in the polygon and after then all the x-coordinates of
+    # polygon points followed by all the y-coordinates.
+    for i in range(polygon_count):
+        line = lines[1 + i]
+        parts = line.split()
+        mask_value = 1
+        index_base = 0
+        if support_mask_values:
+            mask_value = int(parts[0])
+            index_base = 1
+        pnt_count = int(parts[index_base])
+        points = []
+
+        for j in range(0, pnt_count):
+            x = float(parts[index_base + 1 + j])
+            y = float(parts[index_base + 1 + pnt_count + j])
+            points.append((x, y))
+
+        polygons.append((geometry.Polygon(points), mask_value))
```

Comment:
Might consider making this a generator instead of materializing the list of all polygons. Do we need to materialize that list, or will callers be working polygon by polygon, and thus only need an iterator rather than a list?

We'd just do
```python
yield (geometry.Polygon(points), mask_value)
```
here
and then callers would use this function like
```python
for polygon in read_polygons_from_text_file(...)
```
although it'd be more common to name the function `iter_polygons_from_text_file` or such.

Reply:
I don't see any benefits in iteration along the generator over iteration along the normal list. The data in shape files is relatively small comparing to Elevation. Splitting logic that will come a few PRs later keeps all the polygons in memory anyway. Though if we're not satisfied with the memory usage we might reconsider this approach later. But I don't see any reason to spend time on optimization right now.